Rubberduck.Deployment.Build/RubberduckPostBuildTask.cs

```diff
@@ -0,0 +1,365 @@
+            // no longer exist for the current build and thus won't be overwritten)
+            var dir = Path.GetDirectoryName(RegFilePath);
+            if (Directory.Exists(dir))
+            {
+                // The current reg file is now the previous build's reg file
+                var lastRegFile = RegFilePath;
+                if (File.Exists(lastRegFile))
+                {
+                    var now = DateTime.UtcNow;
+                    if (Environment.Is64BitOperatingSystem)
+                    {
+                        var command = $"reg.exe import {lastRegFile} /reg:32";
+                        ExecuteTask(command, ignoreErrors:true);
+                        command = $"reg.exe import {lastRegFile} /reg:64";
+                        ExecuteTask(command, ignoreErrors:true);
+                    }
+                    else
+                    {
+                        var command = $"reg.exe import {lastRegFile}";
+                        ExecuteTask(command, ignoreErrors:true);
+                    }
+                    File.Move(lastRegFile, lastRegFile + ".imported_" + now.ToUniversalTime().ToString("yyyyMMddHHmmss") + ".txt");
+                }
+            }
+            else
+            {
+                Directory.CreateDirectory(dir);
+            }
+
+            _previousRegistrationRemoved = true;
+        }
+
+        private void UpdateDebugRegistration(IOrderedEnumerable<RegistryEntry> entries, DllFileParameters parameters)
+        {
+            this.LogMessage($"Updating debug build reigstration for {parameters.DllFile}");
+
+            // NOTE: The local writer will perform the actual registry changes; the return
+            // is a registry script with deletion instructions for the keys to be deleted
+            // in the next build.
+            var writer = new LocalDebugRegistryWriter();
+            var content = writer.Write(entries, parameters.DllFile, parameters.Tlb32File, parameters.Tlb64File);
+            File.AppendAllText(RegFilePath, content, Encoding.ASCII);
+        }
+
+        private void ExecuteTask(string command, string workingDirectory = null, bool ignoreErrors = false)
+        {
+            var exec = new Exec
+            {
+                BuildEngine = BuildEngine,
+                HostObject = HostObject,
+                Command = command,
+                ConsoleToMSBuild = true,
+                EchoOff = false,
+                LogStandardErrorAsError = !ignoreErrors,
+                IgnoreExitCode = ignoreErrors,
+                UseCommandProcessor = false,
+                WorkingDirectory = workingDirectory ?? ProjectDir,
+                YieldDuringToolExecution = true
+            };
+
+            var log = new TaskLoggingHelper(exec);
+
+            log.LogCommandLine(MessageImportance.Normal, command);
+            var result = exec.Execute();
+
+            if (!result && !log.HasLoggedErrors)
```

Comment:
Do I understand this correctly as a stopgap solution to make sure failing tasks show up in the build log? If yes: Why unconditionally throw an exception here? `ignoreErrors` does imply ignoring the result of the task, right?

If we're ignoring the failure, so long as it logs an error, that seems like a good way to shoot ourselves in the foot.